src/App.tsx

```diff
@@ -15,48 +15,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[], sortType:SortType, isReversed:boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1: string, good2: string) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortAlphabetically = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const sortReverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const reset = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
+
+  const visibleGoods = getReorderedGoods(goodsFromServer, sortType, isReversed);
+  const resetButton = sortType !== SortType.NONE || isReversed !== false;
```

Comment:
You can simplify it for the boolean value 
```suggestion
  const resetButton = sortType !== SortType.NONE || isReversed;
```